pandas/tests/frame/test_constructors.py

```diff
@@ -1336,6 +1336,34 @@ def test_constructor_list_of_namedtuples(self):
         result = DataFrame(tuples, columns=["y", "z"])
         tm.assert_frame_equal(result, expected)
 
+    @pytest.mark.skipif(not PY37, reason="Requires Python >= 3.7")
+    def test_constructor_list_of_dataclasses(self):
+        # GH21910
+        from dataclasses import make_dataclass
+
+        Point = make_dataclass("Point", [("x", int), ("y", int)])
+
+        datas = [Point(0, 3), Point(1, 3)]
+        expected = DataFrame({"x": [0, 1], "y": [3, 3]})
+        result = DataFrame(datas)
+        tm.assert_frame_equal(result, expected)
+
+        # varying types
+        Point = make_dataclass("Point", [("x", int), ("y", int)])
+        HLine = make_dataclass("HLine", [("x0", int), ("x1", int), ("y", int)])
+
+        datas = [Point(0, 3), HLine(1, 3, 3)]
+
+        expected = DataFrame(
+            {"x": [0, np.nan], "y": [3, 3], "x0": [np.nan, 1], "x1": [np.nan, 3]}
+        )
+        result = DataFrame(datas)
+        tm.assert_frame_equal(result, expected)
+
+        # expect TypeError
+        with pytest.raises(TypeError):
```

Comment:
Sorry didn't notice this tucked away here so ignore comment asking to add, but typically for things that raise we create them as a separate test. So would be better to break this off into a test called `test_constructor_dataclass_with_others_raises` or something to the effect

Reply:
I did as asked

Reply:
I think Point isn't defined anymore. Can you move the definition of those to somewhere else? Probably just at the top of this file, not sure.